lib/fs.js

```diff
@@ -688,14 +688,19 @@ fs.renameSync = function(oldPath, newPath) {
 };
 
 fs.truncate = function(path, len, callback) {
-  if (typeof path === 'number') {
+  if (Number.isInteger(path) && path >= 0)
```

Comment:
This is a legacy functionality it seems. If we pass an fd to truncate it internally delegates it to ftruncate. So we cannot throw error here. We are just checking if it is a positive integer and move on to the next check if is not.

Are you suggesting to not to call ftruncate at all?